src/main/scala/nl/knaw/dans/bag/v0/DansV0Bag.scala

```diff
@@ -901,6 +800,29 @@ class DansV0Bag private(private[v0] val locBag: LocBag) extends DansBag {
     recursion(this, src, pathInBag)(mutable.Queue.empty)
   }
 
+  private def movePayloadFile(src: File, pathInData: Path, atomicMove: Boolean): Unit = {
+    val dest = data / pathInData.toString
+    val srcProvider = src.fileSystem.provider()
+
+    if (atomicMove && srcProvider != dest.fileSystem.provider())
+      throw new AtomicMoveNotSupportedException(src.toString(), dest.toString(), s"Different providers, atomic move cannot take place")
+    if (!src.isRegularFile)
+      throw new IllegalArgumentException(s"src cannot be moved, as it is not a regular file: $src")
```

Comment:
I think it is good practice to have `IllegalArgumentException` before anything else. 

Fix the test, don't make the code fit a particular test.

Reply:
It doesn't matter which of these two checks is done first. The provider of a file can be gathered regardless of whether it is a file or folder.

Besides, I don't see a reason why you want the IAE first. I've never seen such rule.